### PR TITLE
feat(CSTM-CPC-854) AddRegisterPage

### PR DESCRIPTION
**Pull request by mattjulienn**
Separating concerns for edit and register pages. Register now has its own page, Edit has its own page per usual.
PLEASE NOTE: 'owners/register' is the new endpoint for registering an owner. 'owners/new' might still work, but 'owners/register' is the new POST URL.

JIRA: https://champlainsaintlambert.atlassian.net/jira/software/c/projects/CPC/boards/14?search=register&selectedIssue=CPC-854

## Context:
Separating concerns for Edit and Register Pages.
'Owner Details' is fully functional and brings you to 'Edit Owner' page, all fully working.
Nav Bar 'Register' brings you to 'Register Owner' page, all fully working.

## Changes
- Added 'Register Owner' page, accessible from Nav Bar. (owners/register instead of owners/new)
- 'Edit Owner' page is no longer accessible via Nav Bar, but only via Edit Owner functionality.

## Before and After UI (Required for UI-impacting PRs)
image
image

## Dev notes (Optional)
- 'owners/register' endpoint is the new 'owners/new' endpoint, please take this into consideration when working on your endpoints.
- BILLS TEAM: I was concerned when trying out the entire PetClinic with the new Register Page. When I checked BILLS and clicked "Owner Details", the form was blank. At first I thought I may have affected it, but I confirmed with two peers who had the most up-to-date version of main and it was like that before my branch updates.

## Linked pull requests (Optional)
- pull request link

### CODE REVIEW

**Review by Nmazza1 — APPROVED** [review]
My man actually did it. Keep in mind you'll need to update your branch with main on the merge!

File: api-gateway/src/main/resources/static/scripts/owner-register/owner-register.controller.js [hunk 1]
```diff
@@ -1,46 +1,46 @@
...
+                self.checked = true
+        }
+
+        self.submitOwnerForm = function () {
+            var id = self.owner.id;
+            console.log(self.owner);
+            var req;
+            if (id){
+                if(method == 'edit')
+                    req = $http.put("api/gateway/owners/" + id, self.owner);
```
Comment (Youssefchahboune) [inline api-gateway/src/main/resources/static/scripts/owner-register/owner-register.controller.js:28]: why do you still have the put and delete in register when the only thing that the register controller / page is doing is a post ?

**Review by Youssefchahboune — CHANGES_REQUESTED** [review]
In your controller make sure you delete the code related to PUT and Delete and only keep what is relevant to adding a new Owner! and don't forget to update your branch ! but apart from that you did it !

File: api-gateway/src/main/resources/static/scripts/owner-register/owner-register.controller.js (hunk 1 above)
Comment (mattjulienn) [reply api-gateway/src/main/resources/static/scripts/owner-register/owner-register.controller.js:28]: youssef. i did this entire page to fulfil your prerequisites you asked of me on wednesday. approve my merge and let me submit. everything works. thanks.

Comment (Nmazza1) [reply api-gateway/src/main/resources/static/scripts/owner-register/owner-register.controller.js:28]: Cleanup and refactoring can be done at a later point, considering the fact he was reusing elements from another aspect of the site.

Comment (mattjulienn) [reply api-gateway/src/main/resources/static/scripts/owner-register/owner-register.controller.js]: @Youssefchahboune Done.

**Review by Nmazza1 — APPROVED** [review]
It's now seperated 👍

**Review by Youssefchahboune — APPROVED** [review]
Looks good now !

**Pull request merged**